scripts/write-config.sh

```diff
@@ -0,0 +1,207 @@
+#   stairway-db-name.txt      | Stairway database name
+#   --------------------------+-------------------------------------------------------------------------
+#   stairway-db-password.txt  | Stairway database password
+#   --------------------------+-------------------------------------------------------------------------
+#   stairway-db-username.txt  | Stairway database username
+#   --------------------------+-------------------------------------------------------------------------
+#   target.txt                | the target that generated this set of config files. Allows the script 
+#                             | to skip regenerating the environment on a rerun.
+#   --------------------------+-------------------------------------------------------------------------
+#   testrunner-sa.json        | SA for running TestRunner
+#   --------------------------+-------------------------------------------------------------------------
+#   user-delegated-sa.json    | Firecloud SA used to masquerade as test users
+#   --------------------------+-------------------------------------------------------------------------
+#   wsm-sa.json               | SA that the WSM application runs as
+#   --------------------------+-------------------------------------------------------------------------
+
+function usage {
+  cat <<EOF
+Usage: $0 [<target>] [<vaulttoken>] [<outputdir>]"
+
+  <target> can be:
+    local - for testing against a local (bootRun) WSM
+    dev - uses secrets from the dev environment
+    alpha - alpha test environment
+    staging - release staging environment
+    help or ? - print this help
+    clean - removes all files from the output directory
+    * - anything else is assumed to be a personal environment using the terra-kernel-k8s
+  If <target> is not specified, then use the envvar WSM_WRITE_CONFIG
+  If WSM_WRITE_CONFIG is not specified, then use local
+
+  <vaulttoken> defaults to the token found in ~/.vault-token.
+
+  <outputdir> defaults to "config/", so when run in the gradle rootdir, it will be
+  in the expected place for automation.
+EOF
+ exit 1
+}
+
+# Get the inputs with defaulting
+default_target=${WSM_WRITE_CONFIG:-local}
+target=${1:-$default_target}
+vaulttoken=${2:-$(cat "$HOME"/.vault-token)}
+outputdir=${3:-config}
+
+# The vault paths are irregular, so we map the target into three variables:
+# k8senv    - the kubernetes environment: alpha, staging, or integration
+# namespace - the namespace in the k8s env: alpha, staging, dev, or the target for personal environments
+# fcenv     - the firecloud delegated service account environment: dev, alpha, staging
+
+case $target in
+    help | ?)
+        usage
+        ;;
+
+    clean)
+        rm "$(pwd)/${outputdir}/*" &> /dev/null
+        exit 0
+        ;;
+
+    local)
+        k8senv=integration
+        namespace=wsmtest
+        fcenv=dev
+        ;;
+
+    dev)
+        k8senv=integration
```

Comment:
Is `k8senv=integration` the intended kubernetes environment for `dev` environment? Isn't `dev` refers to the `DSDE Development` project?

Reply:
Thanks! I didn't realize dev was in a different project. So it follows the alpha and staging pattern.